src/workspacer.Shared/Layout/VerticalTiledEngine.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace workspacer.Shared.Layout
+{
+    public class VerticalTiledEngine : ILayoutEngine
+	{
+        private readonly int _numInPrimary;
+        private readonly double _primaryPercent;
+        private readonly double _primaryPercentIncrement;
+        private readonly int _maxRows;
+
+        private int _numInPrimaryOffset;
+        private double _primaryPercentOffset;
+
+		public string Name => "vertical";
+
+		public VerticalTiledEngine(int numInPrimary, double primaryPercent, double primaryPercentIncrement, int maxRows)
+		{
+			_numInPrimary = numInPrimary;
+			_primaryPercent = primaryPercent;
+			_primaryPercentIncrement = primaryPercentIncrement;
+            _maxRows = maxRows;
```

Comment:
Apologies for the spacing. I have a couple different development environments that don't have synced settings. I'll fix this up.